#! /bin/sh
patch -p1 -f $* < $0
exit $?

From 7b954530dcf20fa60f65cc3c76fe820222697849 Mon Sep 17 00:00:00 2001
From: Markus Montkowski <markus.montkowski@gnupg.com>
Date: Thu, 24 Apr 2025 16:43:14 +0200
Subject: [PATCH] Add Disable addr book check workaround

* common_indep.h: Add disableAddressBookSupport to opts
* main.c: set new disableAddressBookSupport from regisitry
* oomhelp.cpp: log value returned from oom_get_int
* ribbon_callbacks.cpp: return false from
  get_is_addr_book_enabled if new option is set
---
 src/common_indep.h       |  1 +
 src/main.c               |  1 +
 src/oomhelp.cpp          |  2 +-
 src/ribbon-callbacks.cpp | 17 ++++++++++++-----
 4 files changed, 15 insertions(+), 6 deletions(-)

diff --git a/src/common_indep.h b/src/common_indep.h
index 023c121..2d06349 100644
--- a/src/common_indep.h
+++ b/src/common_indep.h
@@ -244,6 +244,7 @@ struct opt_s
   int forms_revision;         /* The forms revision number of the
                                  binary.  */

+  int disableAddressBookSupport; /*Disable addressbook ribon button check*/
 };
 extern struct opt_s opt;

diff --git a/src/main.c b/src/main.c
index 52effe8..b3b3ffe 100644
--- a/src/main.c
+++ b/src/main.c
@@ -269,6 +269,7 @@ read_options (void)
   opt.closeOnUnknownWriteEvent = get_conf_bool ("closeOnUnknownWriteEvent", 0);
   opt.disable_titus_handling = get_conf_bool("disableTitusHandling", 0);
   opt.dont_autodecrypt_preview = get_conf_bool("disableAutoPreview",0);
+  opt.disableAddressBookSupport = get_conf_bool("disableAddressBookSupport",0);

   if (!opt.automation)
     {
diff --git a/src/oomhelp.cpp b/src/oomhelp.cpp
index ceaf652..9d01a3c 100644
--- a/src/oomhelp.cpp
+++ b/src/oomhelp.cpp
@@ -804,7 +804,7 @@ get_oom_int (LPDISPATCH pDisp, const char *name)
         result = rVariant.intVal;
       VariantClear (&rVariant);
     }
-
+  log_debug ("%s:%s: value=%d", SRCNAME, __func__, result);
   TRETURN result;
 }

diff --git a/src/ribbon-callbacks.cpp b/src/ribbon-callbacks.cpp
index 06af440..47ce3e7 100644
--- a/src/ribbon-callbacks.cpp
+++ b/src/ribbon-callbacks.cpp
@@ -611,13 +611,20 @@ HRESULT get_is_details_enabled (LPDISPATCH ctrl, VARIANT *result)

 HRESULT get_is_addr_book_enabled (LPDISPATCH ctrl, VARIANT *result)
 {
+  TSTART;
   if (!ctrl)
    {
      log_error ("%s:%s:%i", SRCNAME, __func__, __LINE__);
-     return E_FAIL;
+     TRETURN E_FAIL;
    }
   result->vt = VT_BOOL | VT_BYREF;
   result->pboolVal = &var_false;
+  if (opt.disableAddressBookSupport)
+    {
+      log_dbg ("Skip addressbook check");
+      TRETURN S_OK;
+    }
+
   LPDISPATCH context = nullptr;
   HRESULT hr = getContext (ctrl, &context);

@@ -625,7 +632,7 @@ HRESULT get_is_addr_book_enabled (LPDISPATCH ctrl, VARIANT *result)
     {
       log_error ("%s:%s:%i :Failed to get context hresult %lx",
                  SRCNAME, __func__, __LINE__, hr);
-      return S_OK;
+      TRETURN S_OK;
     }
   auto selection = get_oom_object_s (context, "Selection");
   gpgol_release (context);
@@ -633,7 +640,7 @@ HRESULT get_is_addr_book_enabled (LPDISPATCH ctrl, VARIANT *result)
     {
       log_error ("%s:%s: Failed to get selection.",
                  SRCNAME, __func__);
-      return S_OK;
+      TRETURN S_OK;
     }
   int count = get_oom_int (selection, "Count");
   if (count == 1)
@@ -646,9 +653,9 @@ HRESULT get_is_addr_book_enabled (LPDISPATCH ctrl, VARIANT *result)
           log_dbg ("One contact selected");
         }

-      return S_OK;
+      TRETURN S_OK;
     }
-  return S_OK;
+  TRETURN S_OK;
 }

 HRESULT get_sig_label (LPDISPATCH ctrl, VARIANT *result)
--
2.39.2
